internal/pkg/addon/addons.go

```diff
@@ -86,17 +93,98 @@ func (a *Addons) Template() (string, error) {
 	return string(out), nil
 }
 
-func filterYAMLfiles(files []string) []string {
-	yamlExtensions := []string{".yaml", ".yml"}
+// Parameters returns the content of user-defined additional CloudFormation Parameters
+// to pass from the parent stack to Template.
+//
+// If there is no addons/ directory defined, then returns "" and ErrAddonsNotFound.
+// If there are addons but no parameters file defined, then returns "" and nil for error.
+// If there are multiple parameters files, then returns "" and cannot define multiple parameter files error.
+// If the addons parameters use the reserved parameter names, then returns "" and a reserved parameter error.
+func (a *Addons) Parameters() (string, error) {
+	fnames, err := a.ws.ReadAddonsDir(a.wlName)
+	if err != nil {
+		return "", &ErrAddonsNotFound{
+			WlName:    a.wlName,
+			ParentErr: err,
+		}
+	}
+	paramFiles := filterFiles(fnames, paramsMatcher)
+	if len(paramFiles) == 0 {
+		return "", nil
+	}
+	if len(paramFiles) > 1 {
+		return "", fmt.Errorf("defining multiple %s is not allowed under %s addons", english.WordSeries(parameterFileNames, "and"), a.wlName)
```

Comment:
Maybe
```suggestion
		return "", fmt.Errorf("defining multiple %s is not allowed under %s addons/", english.WordSeries(parameterFileNames, "and"), a.wlName)
```
Same for the following

Reply:
Done!